### PR TITLE
Add team season

### PR DESCRIPTION
**Pull request by JoshuaTLand**
Added TeamSeason Endpoints and related interfaces / services

Added Postman test to add existing team to an existing season,

### CODE REVIEW

File: sf_stats.Api/sf_stats.Api/appsettings.Development.json [hunk 1]
```diff
@@ -29,14 +29,14 @@
       {
         "Name": "MSSqlServer",
         "Args": {
-          "connectionString": "Server=CHILLTOP-W10;Database=SF_Stats;Trusted_Connection=True;",
+          "connectionString": "Server=JOSHLAPTOP\\SQLEXPRESS;Database=SF_Stats;Trusted_Connection=True;",
```
Comment (GCherry) [inline sf_stats.Api/sf_stats.Api/appsettings.Development.json]: Should probably fix this.... The best option is to use the appsetting.local.json so we don't have to worry about this changing.

File: sf_stats.Api/sf_stats.Api/appsettings.Development.json [hunk 2]
```diff
@@ -29,14 +29,14 @@
...
           "tableName": "Log"
         }
       }
     ]
   },
   "ConnectionStrings": {
-    "DefaultConnection": "Server=CHILLTOP-W10;Database=SF_Stats;Trusted_Connection=True;"
+    "DefaultConnection": "Server=JOSHLAPTOP\\SQLEXPRESS;Database=SF_Stats;Trusted_Connection=True;"
```
Comment (GCherry) [inline sf_stats.Api/sf_stats.Api/appsettings.Development.json]: Same comment as above

File: sf_stats.Api/sf_stats.Domain/DomainObjects/TeamQueryFilter.cs [hunk 3]
```diff
@@ -8,7 +8,7 @@ namespace sf_stats.Domain.DomainObjects
 {
     public class TeamQueryFilter
     {
-        public int Id { get; set; }
+        public int? Id { get; set; }
```
Comment (GCherry) [inline sf_stats.Api/sf_stats.Domain/DomainObjects/TeamQueryFilter.cs:11]: out of curiosity, why does this have to be nullable?

File: sf_stats.Api/sf_stats.Api/appsettings.Development.json (hunk 1 above)
Comment (JoshuaTLand) [reply sf_stats.Api/sf_stats.Api/appsettings.Development.json]: fixed

File: sf_stats.Api/sf_stats.Domain/DomainObjects/TeamQueryFilter.cs (hunk 3 above)
Comment (JoshuaTLand) [reply sf_stats.Api/sf_stats.Domain/DomainObjects/TeamQueryFilter.cs:11]: We want to be able to search for records based without providing an Id

File: sf_stats.Api/sf_stats.Api/appsettings.Development.json (hunk 2 above)
Comment (JoshuaTLand) [reply sf_stats.Api/sf_stats.Api/appsettings.Development.json]: updated

**Review by GCherry — APPROVED** [review]

**Pull request merged**